### ISSUE TITLE
Please provide src code

### ISSUE THREAD

**Issue by jkoenig72** [tevo3d/Tevo-Nereus#2]
Hi,

Still having issues to get auto leveling working.

Can you provide full source code?

Thanks!
Joerg

**Comment by Homers3D** [comment]
Hi Sir , can you please open ticket to us , our engineer will assist you solve the issue .



alex@tlmaker.com
 
From: jkoenig72
Date: 2019-09-21 05:24
To: tevo3d/Tevo-Nereus
CC: Subscribed
Subject: [tevo3d/Tevo-Nereus] Please provide src code (#2)
Hi,
Still having issues to get auto leveling working.
Can you provide full source code?
Thanks!
Joerg

**Comment by xangxiong** [comment]
@jkoenig72 , what I found to work is updating these config values in the robin_nano_cfg.txt and reflashing the firmware.

INVERT_X_DIR 0
INVERT_Y_DIR 0
INVERT_Z_DIR 1
INVERT_E0_DIR 0

Not sure why they were changed before.  Autoleveling seems to work now test movement are working as well.  

If you need the firmware source code, I believe this is using the same MKS Robin Nano here (https://github.com/makerbase-mks/MKS-Robin/tree/master/MKS%20Robin%20Nano).  Firmware generated looks the same, not sure if it is change to fit the Tevo.

**Comment by jkoenig72** [comment]
Thanks!

**Comment by focusedddd** [comment]
The nerus is half baked. Auto leveling is shot. It auto levels, and then on print it runs at the Z probe offset....g29 just keeps pushing Z-axis higher when executing before print. Axis are improperly set in the firmware.

What a joke.

**Comment by xangxiong** [comment]
Yeah, auto leveling is not stable on the Nereus.  I tried the config changes above and the auto leveling seems to work, but is not accurate at all.  First time it auto level, the print turns out fine, but subsequent auto level seems to be slightly off, which resulted in prints not sticking to the bed.  I ended up abandoning the auto leveling completely and just use the manual leveling.

**Comment by focusedddd** [comment]
I just downloaded the robin firmware source, ill fix this and post here the updated firmware.

**Comment by focusedddd** [comment]
at first glance, they have the hotend offset wrong. It should be a negative not positive value

**Comment by focusedddd** [comment]
I fixed it, gonna do a few test prints to confirm its good.

**Comment by deleonjos** [comment]
> I fixed it, gonna do a few test prints to confirm its good.

Can you please provide the steps to fix this issue
